dask/array/tests/test_ghost.py

```diff
@@ -362,3 +363,13 @@ def test_none_boundaries():
          [33, 33,  8,  9, 10, 11, 33, 33],
          [33, 33, 12, 13, 14, 15, 33, 33]])
     assert_eq(exp, res)
+
+
+def test_ghost_small():
+    x = da.ones((10, 10), chunks=(5, 5))
+
+    y = x.map_overlap(lambda x: x, depth=1)
+    assert len(y.dask) < 200
```

Comment:
Currently the algorithm still generates a fair number of aliases that add to this count.  We're not optimal yet.  It would be nice to bring these down, but not necessarily critical.